### ISSUE TITLE
Fix siginfo on risc-v qemu-user

### ISSUE THREAD

**Issue by AntonKozlov** [azul-research/jdk-riscv#29]
Next program behave differently on `qemu-riscv` compared to other options: linux on `qemu-system-riscv`, native linux os, presumably more.

```
#include <signal.h>
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>

int handler(int sig, siginfo_t* info, void* ucVoid) {
    puts("handler()");
    ucontext_t* uc = (ucontext_t*) ucVoid;
    printf("si_addr = 0x%08x\n", info->si_addr);
    printf("si_signo = %d\n", info->si_signo);
    exit(1);
}


int main() {
    struct sigaction act;
    act.sa_sigaction = &handler;
    act.sa_flags = SA_SIGINFO;
    if (sigaction(SIGSEGV, &act, NULL) < 0) {
        perror("sigaction");
        return 1;
    }
    puts("Now I will dereference an invalid pointer");
    int x = *((int*)(0x01));
    return x;
}
```

A bug in `qemu-riscv` is suspected, needs investigation

**Comment by AntonKozlov** [comment]
A workaround: use `qemu-system-riscv` for now

**Comment by tsarn** [comment]
Fixed in QEMU 4.2.0
